sdk/metric/meter_test.go

```diff
@@ -474,3 +475,321 @@ func TestMetersProvideScope(t *testing.T) {
+							Count:        2,
+							Min:          &one,
+							Max:          &two,
+							Sum:          3.0,
+						},
+					},
+					Temporality: metricdata.CumulativeTemporality,
+				},
+			},
+		},
+		{
+			name: "SyncIntCounter",
+			register: func(t *testing.T, mtr metric.Meter) error {
+				ctr, err := mtr.SyncInt64().Counter("sicounter")
+				if err != nil {
+					return err
+				}
+
+				ctr.Add(context.Background(), 10, attribute.Int("version", 1))
+				ctr.Add(context.Background(), 20, attribute.Int("version", 2))
+				return nil
+			},
+			wantMetric: metricdata.Metrics{
+				Name: "sicounter",
+				Data: metricdata.Sum[int64]{
+					DataPoints: []metricdata.DataPoint[int64]{
+						{Value: 30},
+					},
+					Temporality: metricdata.CumulativeTemporality,
+					IsMonotonic: true,
+				},
+			},
+		},
+		{
+			name: "SyncIntUpDownCounter",
+			register: func(t *testing.T, mtr metric.Meter) error {
+				ctr, err := mtr.SyncInt64().UpDownCounter("siupdowncounter")
+				if err != nil {
+					return err
+				}
+
+				ctr.Add(context.Background(), 10, attribute.Int("version", 1))
+				ctr.Add(context.Background(), 20, attribute.Int("version", 2))
+				return nil
+			},
+			wantMetric: metricdata.Metrics{
+				Name: "siupdowncounter",
+				Data: metricdata.Sum[int64]{
+					DataPoints: []metricdata.DataPoint[int64]{
+						{Value: 30},
+					},
+					Temporality: metricdata.CumulativeTemporality,
+					IsMonotonic: false,
+				},
+			},
+		},
+		{
+			name: "SyncIntHistogram",
+			register: func(t *testing.T, mtr metric.Meter) error {
+				ctr, err := mtr.SyncInt64().Histogram("sihistogram")
+				if err != nil {
+					return err
+				}
+
+				ctr.Record(context.Background(), 1, attribute.Int("version", 1))
+				ctr.Record(context.Background(), 2, attribute.Int("version", 2))
+				return nil
+			},
+			wantMetric: metricdata.Metrics{
+				Name: "sihistogram",
+				Data: metricdata.Histogram{
+					DataPoints: []metricdata.HistogramDataPoint{
+						{
+							Bounds:       []float64{0, 5, 10, 25, 50, 75, 100, 250, 500, 750, 1000, 2500, 5000, 7500, 10000},
+							BucketCounts: []uint64{0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+							Count:        2,
+							Min:          &one,
+							Max:          &two,
+							Sum:          3.0,
+						},
+					},
+					Temporality: metricdata.CumulativeTemporality,
+				},
+			},
+		},
+	}
+
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+			v, err := view.New(
+				view.MatchInstrumentName("*"),
+				view.WithFilterAttributes(attribute.Key("foo")),
+			)
+			require.NoError(t, err)
+			rdr := NewManualReader()
+			mtr := NewMeterProvider(WithReader(rdr, v)).Meter("TestAttributeFilter")
+
+			err = tt.register(t, mtr)
+			require.NoError(t, err)
+
+			m, err := rdr.Collect(context.Background())
+			assert.NoError(t, err)
+
+			require.Len(t, m.ScopeMetrics, 1)
+			require.Len(t, m.ScopeMetrics[0].Metrics, 1)
```

Comment:
The exported attributes are not validated to only include the filtered attribute keys. Can this check be added?

Reply:
They are, currently, the exporter set is the empty set.  When I add an attribute to keep, I will also add it to the wantMetric.